kyo-data/shared/src/test/scala/kyo/RecordTest.scala

```diff
@@ -432,10 +432,12 @@ class RecordTest extends Test:
             """)
         }
 
-        "not compile when fields lack CanEqual" in {
+        "not compile when fields lack CanEqual" in pendingUntilFixed { // looks like scala3 bug
```

Comment:
I don't know why this test works in main

When I was fighting with it, I minimized my case to
```scala
import scala.compiletime.*
import scala.language.strictEquality

class Lol
object Lol:
  inline given CanEqual[Lol, Lol] = error("lol")

val x = new Lol
val y = new Lol

// summon[CanEqual[Lol, Lol]] // doesn't compile
x == y // compile, and this is strange
```

https://scastie.scala-lang.org/road21/B8hWnSQNTriBcOYndUduRg

For me this looks like a bug in scala3 compiler, what do you think?

Reply:
yeah, that's odd. I can follow up on this

Reply:
https://github.com/scala/scala3/issues/22420